docker/persistencecore/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21.5-alpine3.18 AS go-builder
+FROM golang:1.22.0-alpine3.18 AS go-builder
```

Comment:
Please don't bump Go version, I wrote that we keep on `golang:1.21.x` until it's upgraded in heighliner there
https://github.com/strangelove-ventures/heighliner/blob/c88e63a992c9cb02c0089e7074acb19248f6fbc7/builder/go_versions.go#L15